### PR TITLE
RATIS-1114. Change NettyServerStreamRpc to use only one ConcurrentMap.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1114

### CODE REVIEW

File: ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java [hunk 1]
```diff
@@ -170,12 +190,12 @@ public void addRaftPeers(Collection<RaftPeer> newPeers) {
...
+  private StreamInfo newStreamInfo(ByteBuf buf) {
     try {
-      final RaftClientRequest request =
-          ClientProtoUtils.toRaftClientRequest(RaftProtos.RaftClientRequestProto.parseFrom(buf.nioBuffer()));
-      return stateMachine.data().stream(request);
-    } catch (InvalidProtocolBufferException e) {
+      final RaftClientRequest request = ClientProtoUtils.toRaftClientRequest(
+          RaftClientRequestProto.parseFrom(buf.nioBuffer()));
+      return new StreamInfo(stateMachine.data().stream(request), proxies.getDataStreamOutput());
+    } catch (Throwable e) {
```
Comment (amaliujia) [inline ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java:198]: Out of curiosity: why not use `Exception e`?

File: ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java [hunk 2]
```diff
@@ -310,26 +324,21 @@ protected void encode(ChannelHandlerContext context, DataStreamReplyByteBuffer r
...
-    workerGroup.shutdownGracefully(0, 100, TimeUnit.MILLISECONDS);
     try {
+      channelFuture.channel().close().sync();
+      bossGroup.shutdownGracefully(0, 100, TimeUnit.MILLISECONDS);
+      workerGroup.shutdownGracefully(0, 100, TimeUnit.MILLISECONDS);
       bossGroup.awaitTermination(1000, TimeUnit.MILLISECONDS);
       workerGroup.awaitTermination(1000, TimeUnit.MILLISECONDS);
     } catch (InterruptedException e) {
-      LOG.error("Interrupt EventLoopGroup terminate", e);
+      LOG.error(this + ": Interrupted close()", e);
```
Comment (amaliujia) [inline ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java:341]: nit: `this + ".close() is Interrupted"`

**Comment by amaliujia** [comment]
Overall looks good to me

**Comment by runzhiwang** [comment]
LGTM

**Comment by runzhiwang** [comment]
@szetszwo Thanks the patch. I have merged it.

**Pull request merged**